### PR TITLE
CONFIGURE: Bring in CFLAGS (v1.2)

### LINKED ISSUE #1547: Stop ignoring CFLAGS request
```
export CFLAGS=-Wno-error
export CXXFLAGS=-Wno-error
./configure CFLAGS=-Wno-error CXXFLAGS=-Wno-error
[   21s] + make -j8 V=1
[   21s] make  all-recursive
[   21s] make[1]: Entering directory '/home/abuild/rpmbuild/BUILD/ucx-0~git2976'
[   21s] Making all in src/ucm
[   21s] make[2]: Entering directory '/home/abuild/rpmbuild/BUILD/ucx-0~git2976/src/ucm'
[   21s] /bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -I/home/a
build/rpmbuild/BUILD/ucx-0~git2976/src -I/home/abuild/rpmbuild/BUILD/ucx-0~git2976/src -DUCM_MALLOC_
PREFIX=ucm_dl  -fno-strict-aliasing -DUSE_LOCKS=1 -DMALLINFO_FIELD_TYPE=int -DCPU_FLAGS="|avx" -Wno-
deprecated-declarations -O3 -g -Wall -Werror  -mavx -c -o event/libucm_la-event.lo `test -f 'event/e
vent.c' || echo './'`event/event.c
```
You *must not* override CFLAGS within configure.ac, or any files included from there. Or if you do, it must be restored at the end of it.

### PR DESCRIPTION
**Pull request by MattBBaker**
Fixes #1547

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1773/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3711/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1775/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1776/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3713/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3714/ for details (Mellanox internal link).

File: config/m4/compiler.m4 [hunk 1]
```diff
@@ -4,9 +4,9 @@
 #
 
 #
-# Initialize CFLAGS
+# Initialize internal CFLAGS
 #
-CFLAGS="-g -Wall -Werror $UCX_CFLAGS"
+AM_CFLAGS="-g -Wall -Werror $UCX_CFLAGS"
```
Comment (miked-mellanox) [inline config/m4/compiler.m4]: do we need to the same for AM_CXXFLAGS?

Comment (MattBBaker) [inline config/m4/compiler.m4]: Kind of? Look into this the black magic just got deeper. I'm going to change this to the previous 'bad version' that still brings in user CFLAGS. Something more comprehensive will be needed to do things 'the right way'.

**Comment by MattBBaker** [comment]
@miked-mellanox Alright, changed it so that it works with C++ now. @boehms Is still right that the configure.ac needs to be fixed up so that it's now rewriting user env variables, but that will require a more comprehensive cleanup of the autoconf stuff.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1779/ for details.

**Comment by shamisp** [comment]
Agree with @boehms and @MattBBaker - it has to be cleaned up. Since we only have 4 libraries, this should not be crazy.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3717/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1800/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3743/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
@yosefe please take a look

**Pull request merged**